src/test/scala/akka/wamp/ManagerSpec.scala

```diff
@@ -56,7 +58,46 @@ class ManagerSpec
     pending
   }
 
-  it should "Fail on wrong message format" in { f =>
+  it should "Receive text messages and emit Events" in { f =>
+    val manager = mockHttpSetup(
+      TextMessage("""[36,1,2,{},[],{"arg0":"paolo","age":40,"arg2":true}]"""),
+      TextMessage("""[36,3,4,{}]""")
+    )
+
+    // connect the router
+    manager ! Connect(client = testActor, url = s"ws://test/ws")
+    expectMsg(Event(1, 2, Dict.apply(), Some(Payload(
+      Map("arg0"->"paolo", "age" -> 40, "arg2" -> true).toList
+    ))))
+    expectMsg(Event(3, 4, Dict.apply(), None))
+  }
+
+  it should "fail on wrong message format" in { f =>
+    val manager = mockHttpSetup(TextMessage("hello world!"))
+
+    // connect the router
+    manager ! Connect(client = testActor, url = s"ws://test/ws")
+    val failure = expectMsgType[Status.Failure](8 seconds)
+    failure.cause.getMessage must include("hello world")
+  }
+
+  it should "use the passed decider" in { f =>
+    val manager = mockHttpSetup(
+      TextMessage("""[36,1,2,{},[],(*&(&({"arg0":"paolo","age":40,"arg2":true}]"""),
+      TextMessage("""[36,3,4,{}]""")
+    )
+
+    val decider: Supervision.Decider = {
```

Comment:
This is cool! Thank you.
This idea to resume the stream and keep it running even though clients send _rubbish messages_ is quite interesting.
I'm posting a message to the [WAMP Forum](https://groups.google.com/forum/#!forum/wampws) to get some feedbacks from the protocol authors and from other developers who are implementing by using other programming languages.